### DISCUSSION TITLE
カラーの指定について

### DISCUSSION THREAD

**Discussion by mcnLeandro** [Recursion-Group-G/Othello#40] [Ideas]
![sshot 3](https://user-images.githubusercontent.com/72016706/140952915-b34d6403-e904-45d3-b3a2-85c2d7a1e08f.png)

**Comment by mcnLeandro** [comment]
![sshot](https://user-images.githubusercontent.com/72016706/140952721-de259ac6-c9df-4fb2-a4f7-507470639f08.png)

**Reply by mcnLeandro** [reply]
![sshot](https://user-images.githubusercontent.com/72016706/140951350-a9ea8d94-6468-49c8-a2ef-3d499cf1dff2.png)

**Reply by mcnLeandro** [reply]
![sshot](https://user-images.githubusercontent.com/72016706/140951606-5cf17150-a33b-48f7-924f-77008607b982.png)

**Reply by mcnLeandro** [reply]
![sshot](https://user-images.githubusercontent.com/72016706/140951655-cf3d25ff-0aaa-460d-b6af-c69c34a87a54.png)

**Reply by mcnLeandro** [reply]
![sshot 2](https://user-images.githubusercontent.com/72016706/140951745-b19c07ac-8627-4346-9cf1-7859cdf53c14.png)

**Reply by mcnLeandro** [reply]
![sshot](https://user-images.githubusercontent.com/72016706/140952091-34594b92-603c-4188-97c5-64ca32725ac2.png)

**Reply by mcnLeandro** [reply]
![sshot](https://user-images.githubusercontent.com/72016706/140952168-b04130e3-b438-43f0-b9de-f043768f21ef.png)

**Reply by mcnLeandro** [reply]
![sshot](https://user-images.githubusercontent.com/72016706/140952277-a5d8f38e-9450-4ed5-b209-abb20969e2a6.png)

**Reply by mcnLeandro** [reply]
![sshot](https://user-images.githubusercontent.com/72016706/140952323-6ed550db-3e5a-427f-bc49-b6b8af0395e8.png)

**Reply by mcnLeandro** [reply]
![sshot](https://user-images.githubusercontent.com/72016706/140952379-3140dc2f-2d41-460e-b69f-b41e053989cf.png)

**Reply by mcnLeandro** [reply]
ここまでがDiscordでの議論でした！
@m2mt-l 
@kuro1221 
@Tucker263 

意見ください！

**Reply by mcnLeandro** [reply]
もう今日は水曜日です。決めないといけないので、意見がなくてもメンターとして決めさせていただきます！
```
- Squareは色を管理すべきでなくStoneを保持しているかだけを管理すべき
- Stoneが色を管理すべき
- 色の型に変更があった際にロジックに影響が出るで対策はしないといけない
- 同じ分野のものを分離すべきでない、また、別の分野のものを同じにすべきではない
```

このことから以下のようにします

`config`
```js
color: {
    white: {
        code: "#ffffff",
        id: 0
    },
    black: {
        code: "#000000",
        id: 1
    }
}
```

`Square`

```js
class Square {
    stone : stone
    isEmpty: boolean //computedを使って管理できる？
}
```

`Stone`

```
class Stone {
    color : {
        code: string,
        id: number
    }
}
```

**Reply by m2mt-l** [reply]
一つ確認なのですが、置ける場所かどうかの判断はこれだとどうすればよいでしょうか？ぱっと思いつく限り以下でしょうか？
```js
class Square {
    stone : stone
    isEmpty: boolean
    isPlaced: boolean
}
```
一旦これで進めますが、やはりこの構造だと判断のロジックの書き方が複雑になるように思えます。
